Lib/shutil.py

```diff
@@ -865,7 +865,10 @@ def _destinsrc(src, dst):
         src += os.path.sep
     if not dst.endswith(os.path.sep):
         dst += os.path.sep
-    return dst.startswith(src)
+    try:
+        return os.lstat(src) == os.lstat(dst[:len(src)])
+    except OSError:
+        return dst.startswith(src)
```

Comment:
In what cases this fallback is needed? And how can it be used if the original issue is that it doesn't work on case-insensitive FS?

Could you please add a test? This is a non-trivial issue and it can have many corner cases.